### PR TITLE
refactor(simplify): streamline planner comparisons and dump checks

### LINKED ISSUE #2222: [simplifier] Go 1.26 modernization and cleanup in planner and dump packages
Code merged in the last 24 hours (PRs #2201, #2208) introduced or
preserved patterns that can be simplified using Go 1.26 stdlib and
general cleanup. All items below are mechanical, behavior-preserving
changes.

---

### 1. Replace `compareStringSlices` with `slices.Equal`

**File:** `internal/declarative/planner/event_gateway_backend_cluster_planner.go` lines 480-489

The manual loop is equivalent to `slices.Equal(a, b)`. Per AGENTS.md
Go 1.26 modernization standards, the `slices` package should be used
instead of manual loops.

```go
// Before
func compareStringSlices(a, b []string) bool {
    if len(a) != len(b) {
        return false
    }
    for i := range a {
        if a[i] != b[i] {
            return false
        }
    }
    return true
}

// After: inline slices.Equal at call sites and remove the function
// Callers in event_gateway_listener_planner.go and
// event_gateway_backend_cluster_planner.go become:
//   slices.Equal(current.Addresses, desired.Addresses)
```

**Callers (3):**
- `event_gateway_backend_cluster_planner.go:365`
- `event_gateway_listener_planner.go:476`
- `event_gateway_listener_planner.go:488`

---

### 2. Replace `compareStringMaps` with `maps.Equal`

**File:** `internal/declarative/planner/event_gateway_backend_cluster_planner.go` lines 512-522

Same rationale. The manual map comparison is `maps.Equal(a, b)`.

```go
// Before
func compareStringMaps(a, b map[string]string) bool { ... }

// After: inline maps.Equal at call sites and remove the function
```

**Callers (2):**
- `event_gateway_backend_cluster_planner.go:406`
- `event_gateway_static_key_planner.go:292`

---

### 3. Remove redundant `reflect.Kind` case listing in `isEmptyValue`

**File:** `internal/cmd/root/verbs/dump/declarative.go` lines 967-986

The exhaustive listing of `reflect.Invalid`, `reflect.Bool`, ...,
`reflect.UnsafePointer` all return `false`, which is identical to the
`return false` default at line 989. The 17-line case block can be
deleted with no behavior change.

---

### 4. Hoist compiled regex out of `isValidHostname`

**File:** `internal/declarative/resources/portal_custom_domain.go` lines 137-143

`regexp.MustCompile` is called on every invocation. Move the compiled
regex to a package-level variable so it is compiled once at init time.

```go
// Before
func isValidHostname(hostname string) bool {
    pattern := `^...`
    hostnameRegex := regexp.MustCompile(pattern)
    return hostnameRegex.MatchString(hostname)
}

// After
var hostnameRegex = regexp.MustCompile(
    `^([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)*` +
        `[a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?$`,
)

func isValidHostname(hostname string) bool {
    return hostnameRegex.MatchString(hostname)
}
```

---

### Validation steps

After applying changes, run:

```sh
go fix ./...
make format
make build
make lint
make test
```

> Generated by [Code Simplifier](https://github.com/Kong/kongctl/actions/runs/35300741437) · opus46 · 236.5 AIC · ⌖ 29.4 AIC · ⊞ 5.9K · [◷](https://github.com/search?q=repo%3AKong%2Fkongctl+is%3Aissue+%22gh-aw-workflow-call-id%3A+Kong%2Fkongctl%2Fcode-simplifier%22&type=issues)
>
<details>
<summary><sub>Add this agentic workflow to your repo</sub></summary>

To install this agentic workflow, run

```
gh aw add githubnext/agentics/workflows/code-simplifier.md@eb7950f37d350af6fa09d19827c4883e72947221
```
</details>

> - [x] expires  on Sep 23, 2026, 2:57 AM UTC

### PR DESCRIPTION
**Pull request by rspurgeon**
Replace the Event Gateway planner's manual slice and map comparisons with
`slices.Equal` and `maps.Equal`, remove redundant reflection cases from dump's
empty-value check, and compile the portal hostname regex once at package init.
The reflection switch uses the existing `exhaustive:ignore` convention because
all unlisted kinds intentionally return false.

`maps.Equal` also detects changes between different label keys with empty
values, which the old helper incorrectly treated as equal. A static-key planner
regression test covers this edge case. The separate order-independent API
planner comparison is unchanged.

Validation: `CGO_ENABLED=0 go fix ./...`, `make format` (no changes), `make build`,
`make lint`, `make test`, and `make test-all` passed, including race-enabled unit,
E2E harness, and integration tests. The full suite ran without the unconfigured
mise shim directory on PATH; optional shellcheck was skipped because it is not
installed. No E2E scenarios or recordings changed.

Closes #2222.

### CODE REVIEW

File: internal/declarative/planner/event_gateway_backend_cluster_planner.go [hunk 1]
```diff
@@ -403,7 +404,7 @@ func (p *Planner) shouldUpdateBackendCluster(
 
 	// Compare labels (user labels only, ignore KONGCTL labels)
 	if desired.Labels != nil {
-		if !compareStringMaps(current.Labels, desired.Labels) {
+		if !maps.Equal(current.Labels, desired.Labels) {
```
[the remark this answers is not in the document]
Comment (rspurgeon) [reply internal/declarative/planner/event_gateway_backend_cluster_planner.go:407]: Thanks for catching the missing coverage. Added TestShouldUpdateBackendCluster_EmptyLabelKeyChanged in 194f2c9e. It verifies that swapping empty-valued label keys requests an update, includes the desired labels in the update payload, and reports exactly the labels field change with the old and new maps. Matching anonymous authentication prevents unrelated changes from masking a regression. Verified with go fix, make format, make build, make lint, make test, and make test-all (including integration tests).
